File: Material Core/src/main/java/material/core/MaterialLayoutInflater.java
package material.core;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;

public class MaterialLayoutInflater extends LayoutInflater {
    protected MaterialLayoutInflater(Context context) {
        super(context);
    }

    protected MaterialLayoutInflater(LayoutInflater original, Context newContext) {
        super(original, newContext);
    }

    @Override
    public LayoutInflater cloneInContext(Context newContext) {
        return new MaterialLayoutInflater(this, newContext);
    }

    @Override
    protected View onCreateView(String name, AttributeSet attrs) throws ClassNotFoundException {
        if (Character.isUpperCase(name.charAt(0)))
            return createView(name, "carbon.widget.", attrs);
        return super.onCreateView(name, attrs);
    }

    @Override
    protected View onCreateView(View parent, String name, AttributeSet attrs) throws ClassNotFoundException {
        if (Character.isUpperCase(name.charAt(0)))
            return createView(name, "carbon.widget.", attrs);
        return super.onCreateView(parent, name, attrs);
    }
}

File: core/src/main/java/material/core/MaterialLayoutInflater.kt
package material.core

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View

/**
 * Created by ali on 7/3/2018.
 */
class MaterialLayoutInflater(original: LayoutInflater?, context: Context?) : LayoutInflater(original, context) {
    override fun cloneInContext(context: Context?): LayoutInflater = MaterialLayoutInflater(this, context)
    override fun onCreateView(name: String, attrs: AttributeSet): View =
            if (name[0].isUpperCase() ) createView(name, "mat.widget.", attrs) else super.onCreateView(name, attrs)

    override fun onCreateView(parent: View?, name: String, attrs: AttributeSet): View =
            if (name[0].isUpperCase()) createView(name, "mat.widget.", attrs) else super.onCreateView(parent, name, attrs)
}